juniper_official/Kernel/check-stuck-ifstate-clients.rule

```diff
@@ -0,0 +1,53 @@
+/*
+ * This rule checks if there are any stuck ifstate clients present in the system.
+ *
+ * A stuck Ifstate client can cause resource release issues and eventually lead to
+ * Kernel to veto incoming routing socket messages from applications, which further
+ * can result in traffic loss. Though very rare, if this happens, it is a serious
+ * condition and should be mitigated asap by restarting the client. See readme
+ * in healthbot rules folder for details.
+ */
+healthbot {
+    topic kernel.ifstate {
+        rule check-stuck-ifstate-clients {
+            sensor ifstate-clients {
+                synopsis "stuck ifstate client open-config sensor";
+                description "open-config sensor to monitor any stuck ifstate client on a network device";
+                open-config {
+                    sensor-name /junos/kernel-ifstate;
+                    frequency 1m;
+                }
+            }
+            field stuck-clients {
+                sensor ifstate-clients {
+                    path /junos/kernel-ifstate/stuck-clients-cnt;
+                }
+                type integer;
+            }
+            trigger report-ifstate-clients-health {
+                synopsis "report stuck ifstate clients";
+                description "monitor and report any stuck ifstate clients on the network device";
+                frequency 90s;
+                term is-ifs-client-stuck {
+                    when {
+                        not-equal-to "$stuck-clients" 0;
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "Found stuck daemon/fpc. For details run: show system alarms. Restart stuck clients for recovery.";
```

Comment:
Would it make sense to collect the output of "show system alarm" through another rule and just reference it here in the message - stating that here is the daemon/fpc that is stuck"?

Reply:
Checked with prateek. It is difficult to predict fpc or daemon and also not possible to get information from other rules as it is based on event.

Reply:
Checked with prateek. It is difficult to predict fpc or daemon and also not possible to get information from other rules as it is based on event.